pkg/apis/apps/v1alpha1/servicebindingrequest_types.go

```diff
@@ -52,6 +52,8 @@ type CustomEnvMap struct {
 type ServiceBindingRequestStatus struct {
 	// BindingStatus is the status of the service binding request.
 	BindingStatus string `json:"bindingStatus,omitempty"`
+	// Reason indicates the reason for any failures
+	Reason string `json:"reason,omitempty"`
```

Comment:
Curious, is `reason` a standard attribute?

Reply:
I feel naming should `FailureReason` or we could have separate `Failure` struct to populate more information.

Reply:
@sbose78 @akashshinde I referred this - https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties

Reply:
I find the convention `status.conditions` from other Kubernetes types quite convenient, and users already know where to look and the meaning of its contents. Other than that, it can be used by any UI to present SBO's conditions since its data structure is quite simple.

Reply:
Yeah, I've seen status.conditions very useful in a lot of places. Let's use that, if that's ok, Dhriti?

Reply:
```
type ConditionStatus string

const (
    ConditionTrue    ConditionStatus = "True"
    ConditionFalse   ConditionStatus = "False"
    ConditionUnknown ConditionStatus = "Unknown"
)
```

> These are valid condition statuses. "ConditionTrue" means a resource is in the condition. "ConditionFalse" means a resource is not in the condition. "ConditionUnknown" means kubernetes can't decide if a resource is in the condition or not.

Refer https://godoc.org/k8s.io/api/core/v1#ConditionStatus

ConditionStatus is just about the status of the resource. 
This task talks about adding failure message/reason in status message to help debug - https://github.com/redhat-developer/service-binding-operator/issues/238

Reply:
@isutton @sbose78 @akashshinde Please review this PR